test/templates/templates_register_api_test.rb

```diff
@@ -0,0 +1,26 @@
+require 'test_helper'
+require 'templates/templates_register_api'
+require 'templates/templates'
+require 'webmock/test_unit'
+
+class TemplatesRegisterApiTest < Test::Unit::TestCase
+  include Rack::Test::Methods
+
+  def app
+    Proxy::TemplatesRegisterApi.new
+  end
+
+  def setup
+    @foreman_url = 'http://foreman.example.com'
+    Proxy::SETTINGS.stubs(:foreman_url).returns(@foreman_url)
+    @template_url = 'http://foreman.example.com:8000'
```

Comment:
I wonder why you didn't use my suggestion. The reason I suggested it is that in a typical deployment you have a machine that runs Foreman on `foreman.example.com`. The Smart Proxy can run on the same machine, but can also run on another machine. My suggestion to use `smart-proxy.example.com` as a hostname was to clearly indicate that there doesn't have to be any relation between `foreman_url` and `template_url`.

This is a good practice to keep. If there is any code that parses URLs and rewrites, it will fail. That is a good thing. There's a reason you shouldn't use `@template_url = "#{foreman_url}:8000"`.

Reply:
I missed that somehow, changed to smart-proxy.example.com